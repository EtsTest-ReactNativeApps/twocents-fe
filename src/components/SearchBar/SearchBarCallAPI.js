import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';

export class SearchBarCallAPI extends Component {
  constructor(props) {
    super(props)
    this.state ={
      name: '', 
      address: 'Current Location',
      error: ''
    }
  }

  handleChangeName = enteredText => {
    this.setState({name: enteredText})
  } 
  
  handleChangeAddress = enteredText => {
    this.setState({address: enteredText})
  }

  handleSubmit = () => {
    if(!this.state.name || !this.state.address) {
      this.setState({error: 'Missing required field'})
    } else {
      if(this.state.address === 'Current Location') {
       console.log(this.props.userLocation)
      } else {
       console.log(this.state.address)
      }
    }
    this.setState({name: '', address: 'Current Location'})
  }

  render() {
    return (
      <View style={styles.container}>
          <TextInput 
            placeholder='Enter Recommended Location'
            style={styles.input}
            value={this.state.name}
            onChangeText={this.handleChangeName}
            onSubmitEditing={this.handleSubmit}
          />
          <TextInput 
            placeholder='Enter City and State'
            style={styles.input}
            value={this.state.address}
            onChangeText={this.handleChangeAddress}
            onSubmitEditing={this.handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    height: 70,
    width: '84%',
    borderRadius: 20,
  },
  input: {
    width: '90%',
    color: '#FFF',
  }
});

const mapStateToProps = state => ({
  userLocation: state.userLocation
})

export default connect(mapStateToProps)(SearchBarCallAPI);