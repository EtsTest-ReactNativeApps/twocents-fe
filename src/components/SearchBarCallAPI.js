import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';

class SearchBarCallAPI extends Component {
  constructor(props) {
    super(props)
    this.state ={
      place: '', 
      location: ''
    }
  }

  render() {
    const handleChangePlace = enteredText => {
      this.setState({place: enteredText})
    } 
    
    const handleChangeLocation = enteredText => {
      this.setState({location: enteredText})
    }
    const handleSubmit = () => {
      console.log(this.state)
      this.setState({place: '', location: ''})
    }
    return (
      <View style={styles.container}>
          <TextInput 
            placeholder='Enter Recommended Location'
            style={styles.input}
            value={this.state.place}
            onChangeText={handleChangePlace}
            onSubmitEditing={handleSubmit}
          />
          <TextInput 
            placeholder='Enter City and State'
            style={styles.input}
            value={this.state.location}
            onChangeText={handleChangeLocation}
            onSubmitEditing={handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    height: 70,
    width: '84%',
    borderRadius: 20,
  },
  input: {
    width: '90%',
    color: '#FFF',
  }
});

export default SearchBarCallAPI;