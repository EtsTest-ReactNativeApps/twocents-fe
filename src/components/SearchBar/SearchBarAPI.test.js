import React from 'react';
import { shallow } from 'enzyme';
import { SearchBarCallAPI } from './SearchBarCallAPI';

describe('SearchBarCallAPI', () => {

  it('should match to snapshot', () => {
    const component = shallow(<SearchBarCallAPI/>)
    expect(component).toMatchSnapshot()
  });
  
});