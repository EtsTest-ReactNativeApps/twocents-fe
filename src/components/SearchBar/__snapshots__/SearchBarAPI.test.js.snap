// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarCallAPI should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(204, 192, 221, 0.4)",
      "borderRadius": 20,
      "height": 70,
      "justifyContent": "space-around",
      "width": "84%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter Recommended Location"
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "color": "#FFF",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter City and State"
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "color": "#FFF",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value="Current Location"
  />
</View>
`;
